src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,119 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationFailedException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RegistrationServiceImplTest {
+    private StorageDaoImpl storageDao;
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+    }
+
+    @Test
+    void nullUser_NotOK() {
+        assertThrows(RegistrationFailedException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void nullLogin_NotOK() {
+        User user = buildUser(null, "123456", 19);
+        assertThrows(RegistrationFailedException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void nullPassword_NotOK() {
+        User user = buildUser("SampleLogin", null, 19);
+        assertThrows(RegistrationFailedException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void nullAge_NotOK() {
+        User user = buildUser("SampleLogin", "123456", null);
+        assertThrows(RegistrationFailedException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void oneUser_OK() {
+        User user = buildUser("SampleLogin11111", "123456", 22);
+        registrationService.register(user);
+        String login = user.getLogin();
+        assertEquals(user, storageDao.get(login));
+    }
```

Comment:
Make sure that after successful registration user was added to storage and the `register` method returned the correct user. This test checks if the user is in storage, but it doesn't verify the return value of the `register` method.